--- kernel/nv-linux.h   2013-09-03 22:46:51.397145404 -0400
+++ kernel/nv-linux.h   2013-09-03 22:53:45.112478362 -0400
@@ -956,11 +956,15 @@ static inline int nv_execute_on_all_cpus
         *(mutex) = __mutex;                        \
     }
 #endif
 
 #if !defined(NV_VMWARE)
+#if LINUX_VERSION_CODE < KERNEL_VERSION(3, 11, 0)
 #define NV_NUM_PHYSPAGES                num_physpages
+#else
+#define NV_NUM_PHYSPAGES                (get_num_physpages())
+#endif
 #define NV_GET_CURRENT_PROCESS()        current->tgid
 #define NV_IN_ATOMIC()                  in_atomic()
 #define NV_LOCAL_BH_DISABLE()           local_bh_disable()
 #define NV_LOCAL_BH_ENABLE()            local_bh_enable()
 #define NV_COPY_TO_USER(to, from, n)    copy_to_user(to, from, n)
