--- a/src/reconfigure.c	2012-03-01 17:38:19.382944425 +0400
+++ b/src/reconfigure.c	2012-03-01 17:36:12.865014080 +0400
@@ -592,7 +592,7 @@
 
     ok_permission = 0;
     if (fstat(fileno(file), &st) == 0)
-	ok_permission = ((st.st_uid == getuid()) && ((st.st_mode & 0177) == 0));
+	ok_permission = (/*(st.st_uid == getuid()) && */((st.st_mode & 0117) == 0));
     else DPRINTF(3, "Stat file %s error : %s\n", filename, strerror(errno));
 
     fscanf(file, "%[\n]", s);
