--- ext/gd/gd.c	2014-03-05 14:18:00.000000000 +0400
+++ ext/gd/gd.c	2014-04-04 18:19:26.244605684 +0400
@@ -4879,7 +4879,7 @@
 				if (zend_hash_index_find(Z_ARRVAL_PP(var), (j), (void **) &var2) == SUCCESS) {
 					if (Z_TYPE_PP(var2) != IS_DOUBLE) {
 						zval dval;
-						dval = **var;
+						dval = **var2;
 						zval_copy_ctor(&dval);
 						convert_to_double(&dval);
 						matrix[i][j] = (float)Z_DVAL(dval);
