--- a/python/mainwindow.py
+++ b/python/mainwindow.py
@@ -240,7 +240,6 @@
 
         ## List game
         self.list_game = wx.TreeCtrl(self, 105, style=wx.TR_HIDE_ROOT | wx.TR_FULL_ROW_HIGHLIGHT)
-        self.list_game.SetSpacing(0)
         self.list_game.SetIndent(5)
         self.list_game.SetImageList(self.images)
         self.menu_gauche = wx.Panel(self, -1)

--- a/python/install/InstallWindow.py
+++ b/python/install/InstallWindow.py
@@ -183,7 +183,6 @@
         self.appsList = wx.TreeCtrl(self.appNavigationPanel, 106,
                                     style=wx.TR_HIDE_ROOT | wx.TR_FULL_ROW_HIGHLIGHT | Variables.widget_borders)
         self.appsList.SetImageList(self.imagesapps)
-        self.appsList.SetSpacing(0)

         self.appNavigationSizer.AddSpacer(10)
         self.appNavigationSizer.Add(self.appsList, 4, wx.EXPAND, 5)

--- a/python/setupwindow/POL_SetupFrame.py
+++ b/python/setupwindow/POL_SetupFrame.py
@@ -212,7 +212,6 @@
                                      style=wx.TR_HIDE_ROOT | wx.TR_FULL_ROW_HIGHLIGHT | Variables.widget_borders,
                                      pos=(25, 25), size=(460, 220))
         self.MenuGames.SetImageList(self.images)
-        self.MenuGames.SetSpacing(0)

         # Login

--- a/python/debug.py
+++ b/python/debug.py
@@ -60,7 +60,6 @@
         self.Bind(wx.EVT_TREE_SEL_CHANGED, self.analyseLog, id=900)


-        self.list_game.SetSpacing(0);
         self.list_game.SetImageList(self.images)


--- a/python/wine_versions/WineVersionsNotebook.py
+++ b/python/wine_versions/WineVersionsNotebook.py
@@ -39,7 +39,6 @@
         self.installedWineVersionsTreeSelector[arch] = wx.TreeCtrl(installedWineVersionsPanel, 107 + self.calculateEventDelta(arch),
                                                                    style=wx.TR_HIDE_ROOT | wx.TR_FULL_ROW_HIGHLIGHT | Variables.widget_borders)
         self.installedWineVersionsTreeSelector[arch].SetImageList(self.imagesapps_i[arch])
-        self.installedWineVersionsTreeSelector[arch].SetSpacing(0)
         installedWineVersionsSizer.Add(self.installedWineVersionsTreeSelector[arch], 1, wx.EXPAND)
         installedWineVersionsSizer.AddSpacer(10)


--- a/python/wine_versions/WineVersionsNotebook.py
+++ b/python/wine_versions/WineVersionsNotebook.py
@@ -55,7 +54,6 @@
                                                                    style=wx.TR_HIDE_ROOT | wx.TR_FULL_ROW_HIGHLIGHT | Variables.widget_borders,
                                                                    size=(320, 300), pos=(10, 35))
         self.availableWineVersionsTreeSelector[arch].SetImageList(self.imagesapps[arch])
-        self.availableWineVersionsTreeSelector[arch].SetSpacing(0)
         availableWineVersionsSizer.Add(self.availableWineVersionsTreeSelector[arch], 1, wx.EXPAND)
         availableWineVersionsSizer.AddSpacer(10)


--- a/python/options.py
+++ b/python/options.py
@@ -187,7 +187,6 @@
         self.txtPlugin = wx.StaticText(self.panelPlugins, -1, _("Installed plugins"), size=wx.DefaultSize)
         self.txtPlugin.SetFont(self.fontTitle)
         self.pluginlist = wx.TreeCtrl(self.panelPlugins, 220, style=Variables.widget_borders|wx.TR_HIDE_ROOT|wx.TR_FULL_ROW_HIGHLIGHT)
-        self.pluginlist.SetSpacing(0)

         self.pluginImgList = wx.ImageList(16,16)

--- a/python/configurewindow/ConfigureWindowNotebook.py
+++ b/python/configurewindow/ConfigureWindowNotebook.py
@@ -269,7 +269,6 @@

         self.Menu = wx.TreeCtrl(self.panelPackages, 99, pos=(15, 75), size=(530, 260),
                                 style=wx.TR_HIDE_ROOT | wx.TR_FULL_ROW_HIGHLIGHT | Variables.widget_borders)
-        self.Menu.SetSpacing(0);
         self.Menu.SetImageList(self.imagePackages)
         self.imagePackages.RemoveAll()

--- a/python/configurewindow/ConfigureWindow.py
+++ b/python/configurewindow/ConfigureWindow.py
@@ -75,7 +75,6 @@
         self.Bind(wx.EVT_BUTTON, self.DeletePrefix, id=1002)


-        self.list_game.SetSpacing(0)
         self.list_game.SetImageList(self.images)


--- a/python/lib/playonlinux.py
+++ b/python/lib/playonlinux.py
@@ -278,11 +278,11 @@
         else:
             return False
 
-    version1 = [ int(digit) for digit in version1[0].split(".") ]
+    version1 = [ int(digit if digit.isdecimal() else 0) for digit in version1[0].split(".") ]
     while len(version1) < 3:
         version1.append(0)
 
-    version2 = [ int(digit) for digit in version2[0].split(".") ]
+    version2 = [ int(digit if digit.isdecimal() else 0) for digit in version2[0].split(".") ]
     while len(version2) < 3:
         version2.append(0)

