diff -Nur xbmc-14.1-Helix.orig/configure.in xbmc-14.1-Helix.new/configure.in
--- xbmc-14.1-Helix.orig/configure.in	2015-01-29 23:57:10.000000000 +0100
+++ xbmc-14.1-Helix.new/configure.in	2015-03-30 22:08:00.661976231 +0200
@@ -1206,9 +1206,6 @@
     AC_DEFINE([SDL_VERSION],[1],["SDL major version"])
   fi
 else
-if test "$target_platform" != "target_android" ; then
-  AC_CHECK_LIB([jasper],     [main],, AC_MSG_ERROR($missing_library)) # check for cximage
-fi
 if test "x$use_alsa" != "xno"; then
   PKG_CHECK_MODULES([ALSA],  [alsa],
     [INCLUDES="$INCLUDES $ALSA_CFLAGS"; LIBS="$LIBS $ALSA_LIBS"; use_alsa=yes],
diff -Nur xbmc-14.1-Helix.orig/lib/cximage-6.0/Makefile.in xbmc-14.1-Helix.new/lib/cximage-6.0/Makefile.in
--- xbmc-14.1-Helix.orig/lib/cximage-6.0/Makefile.in	2015-01-29 23:57:10.000000000 +0100
+++ xbmc-14.1-Helix.new/lib/cximage-6.0/Makefile.in	2015-03-30 21:38:27.188665277 +0200
@@ -4,8 +4,8 @@
 ARCHIVE=$(SO:.so=.a)
 SYSDIR=../../system
 SLIB=$(SYSDIR)/$(SO)
-DIRS=CxImage raw jbig
-SLIBS=CxImage/cximage.a raw/raw.a jbig/jbig.a
+DIRS=CxImage raw
+SLIBS=CxImage/cximage.a raw/raw.a
 
 ifeq ($(findstring osx,$(ARCH)), osx)
 	export ARCH
@@ -21,7 +21,7 @@
 		CxImage/*.o jasper/*/*.o jbig/*.o raw/*.o
 else
 	$(CXX) $(CFLAGS) $(LDFLAGS) -shared -o $(SLIB) -Wl,--whole-archive $(SLIBS) \
-		-Wl,--no-whole-archive -lpng -ljasper -ljpeg -ltiff -lz -Wl,--unresolved-symbols=ignore-all
+		-Wl,--no-whole-archive -lpng -ljpeg -ltiff -lz -Wl,--unresolved-symbols=ignore-all
 endif
 
 $(SLIBS): compile 

