diff -Nur xbmc-14.1-Helix.orig/bootstrap.mk xbmc-14.1-Helix.new/bootstrap.mk
--- xbmc-14.1-Helix.orig/bootstrap.mk	2015-03-29 13:38:25.594436410 +0200
+++ xbmc-14.1-Helix.new/bootstrap.mk	2015-03-29 13:39:03.601098833 +0200
@@ -2,7 +2,7 @@
 BOOTSTRAP_SUBDIRS += xbmc/screensavers/rsxs-0.9/configure.ac
 BOOTSTRAP_SUBDIRS += xbmc/visualizations/Goom/goom2k4-0/configure.in
 BOOTSTRAP_SUBDIRS += lib/cpluff/configure.ac
-BOOTSTRAP_SUBDIRS += lib/gtest/configure.ac
+#BOOTSTRAP_SUBDIRS += lib/gtest/configure.ac
 
 ifneq ($(wildcard lib/libdvd/libdvdcss/configure.ac),)
 BOOTSTRAP_SUBDIRS += lib/libdvd/libdvdcss/configure.ac

